src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -386,8 +392,20 @@ public void StartProfileOptimization(string profile)
             InternalStartProfile(profile, m_pNativeAssemblyLoadContext);
 #endif // FEATURE_MULTICOREJI
         }
-        
+
+        private void OnProcessExit(object sender, EventArgs e)
+        {
+            var unloading = Unloading;
+            if (unloading != null)
+            {
+                unloading(this, EventArgs.Empty);
+            }
+        }
+
         public event Func<AssemblyLoadContext, AssemblyName, Assembly> Resolving;
+        public event EventHandler Unloading;
```

Comment:
Have the API reviewers looked at the signature of this API? It looks odd to see two events next to each - one that is `EventHandler` and other that is `Func` ... but maybe it is ok.